pkg/generate/template_test.go

```diff
@@ -0,0 +1,114 @@
+package generate
+
+import (
+	"io"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestHas(t *testing.T) {
+	for _, tt := range []struct {
+		name     string
+		file     io.Reader
+		find     string
+		expected bool
+	}{
+		{
+			name: "can find a value if present",
+			file: strings.NewReader(`---
+testOne: 123
+---
+testTwo: 456
+`),
+			find:     "testOne: 123",
+			expected: true,
+		},
+		{
+			name: "does not find a value if not present",
+			file: strings.NewReader(`---
+testOne: 123
+---
+testTwo: 456
+`),
+			find:     "testThree: 789",
+			expected: false,
+		},
+		{
+			name:     "responds correctly given an empty template",
+			file:     strings.NewReader(""),
+			find:     "testThree: 789",
+			expected: false,
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			actualTemplate, err := buildTemplate(tt.file)
+			if err != nil {
+				t.Fatal("error with test, not able to create template:", err)
+			}
+
+			actual := actualTemplate.has(tt.find)
+			require.Equal(t, tt.expected, actual)
+		})
+	}
+}
+
+func TestWith(t *testing.T) {
```

Comment:
Do we care about cases where there's a key in the map but it doesn't exist in the template?

Reply:
We do for sure, added a test case for this.